src/common/hooks/useSupportsInterface.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from "react";
```

Comment:
Isnt this file just a rewrite of ethers-contract-hook?

Reply:
it's a wrapper around ethers-contract-hook that is supposed to encapsulate the logic of interpreting the various results from calling supportsInterface, but it looks like intern didn't quite get that interpretation part..........

1) supportsInterface returns true = it is a smart contract that supports xx interface
2) supportsInterface returns false = it is a smart contract that has Erc165 but does not support xx interface
3) supportsInterface throws error of method not supported = it is a smart contract but does not have Erc165
4) supportsInterface throws error of contract not deployed = address is not a wallet

in cases 2-4 we can simply return the state of interface not supported instead of leaving the caller to figure out what's going on

Reply:
yea that will be more appropriate for true/false to be returned, unless there is meaning to determine which of the 4 types there are.. then its not exactly a generic 'useSupportInterface' but 'useOwnerType'